crates/bevy_ecs/src/system/query.rs

```diff
@@ -1761,6 +2011,21 @@ impl<'w, 's, D: QueryData, F: QueryFilter> Query<'w, 's, D, F> {
     }
 }
 
+impl<'w, 's, D: QueryData, F: QueryFilter> IntoIterator for Query<'w, 's, D, F> {
+    type Item = D::Item<'w>;
+    type IntoIter = QueryIter<'w, 's, D, F>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        // SAFETY:
+        // - `self.world` has permission to access the required components.
+        // - The query is read-only, so it can be aliased even if it was originally mutable.
```

Comment:
I'm probably missing something but why is this read-only? The `QueryData` isn't limited and `Item` is the one from `WorldQuery` which would be mutable for a mutable query.

Reply:
Ah, this safety comment comes from `Query::iter`, it should probably take the one from `Query::iter_mut` instead which doesn't mention the query being read-only. This is sound because we take `self` so you can't be holding another `&Query` to iterate over and potentially alias.

Reply:
Yup, that safety comment is wrong.  I must have copied it from a read-only version.  I'll change it to "We consume the query, so mutable queries cannot alias.  Read-only queries are `Copy`, but may alias themselves."